tools/aot/test_aot_compiled.template.cpp

```diff
@@ -0,0 +1,137 @@
+//===------------------------------------------------------------*- C++ -*-===//
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// Also available under a BSD-style license. See LICENSE.
+//
+//===----------------------------------------------------------------------===//
+
+#include "tools/aot/abi.h"
+
+#include "gtest/gtest.h"
+
+using namespace mlir::tcp;
+
+#pragma clang diagnostic ignored "-Wreturn-type-c-linkage"
+
+extern "C" RankedMemref<float, 2> func_1(DECL_RANK_2_MEMREF_ABI(float),
+                                       DECL_RANK_2_MEMREF_ABI(float),
+                                       DECL_RANK_2_MEMREF_ABI(float));
```

Comment:
Currently hitting an error because the function name defaults to `main` in the torch exporter.

```
ERROR: /home/sambhav.jain/workspaces/mlir-tcp/src/github.com/cruise-automation/mlir-tcp/test/AotCompile/BUILD:21:12: Linking test/AotCompile/test_aot_compiled_add_mul_net_single_output failed: (E
xit 1): clang failed: error executing command (from target //test/AotCompile:test_aot_compiled_add_mul_net_single_output) /home/sambhav.jain/bin/clang @bazel-out/k8-fastbuild/bin/test/AotCompile/
test_aot_compiled_add_mul_net_single_output-2.params                                                                                                                                               
                                                                                                                                                                                                   
Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging                                                                                       
ld.lld: error: undefined symbol: func_1                      
```

This [upstream PR](https://github.com/llvm/torch-mlir/pull/2949) updates the fx importer API to support changing `func_name`.